src/main/java/ghidra/emotionengine/MipsR5900PreAnalyzer.java

```diff
@@ -0,0 +1,422 @@
+/* ###
+ * IP: GHIDRA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ghidra.emotionengine;
+
+import java.math.BigInteger;
+
+import ghidra.app.services.*;
+import ghidra.app.util.importer.MessageLog;
+import ghidra.program.disassemble.Disassembler;
+import ghidra.program.model.address.*;
+import ghidra.program.model.lang.Processor;
+import ghidra.program.model.lang.Register;
+import ghidra.program.model.listing.*;
+import ghidra.program.model.mem.MemoryAccessException;
+import ghidra.program.model.mem.MemoryBlock;
+import ghidra.program.model.scalar.Scalar;
+import ghidra.program.model.symbol.FlowType;
+import ghidra.util.Msg;
+import ghidra.util.exception.CancelledException;
+import ghidra.util.task.TaskMonitor;
+
+public class MipsR5900PreAnalyzer extends AbstractAnalyzer {
```

Comment:
does this need to be copy-pasted btw? seems maybe you could just inherit `MipsR5900PreAnalyzer` and override `canAnalyze`

Reply:
I think that's a possibility, but there's two caveats:

- The upstream Ghidra version requires several extra context registers to exist in the SLEIGH definition for compatibility with several different MIPS ISAs, which the R5900 doesn't have to deal with. I removed the code for those ISAs here.
- The analyzer relies on hard-coded `prime` opcodes which exactly correspond to the relevant MIPS instructions (`lwr`, `swr`, ...). If upstream Ghidra changes the prime number of any of these opcodes in a future update and we inherit from their analyzer, our analyzer will break.

I think the possibility of the latter is slim, but still worth considering.